firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/common/CrashlyticsCoreTest.java

```diff
@@ -102,11 +103,13 @@ public void testCustomAttributes() throws Exception {
     final String superLongId = longId + "more chars";
 
     crashlyticsCore.setUserId(superLongId);
+    Thread.sleep(10);
     assertEquals(longId, metadata.getUserId());
 
     final String key1 = "key1";
     final String value1 = "value1";
     crashlyticsCore.setCustomKey(key1, value1);
+    Thread.sleep(10);
```

Comment:
Since we move all user action to background, how do we want to fix the unit tests?

- moving assertion to background thread, await on it
- do Thread.sleep everytime if we know it gonna run on background thread.
- inplement getter as task and return future